src/vs/workbench/services/progress/browser/progressService.ts

```diff
@@ -376,18 +383,38 @@ export class ProgressService extends Disposable implements IProgressService {
 		// show in viewlet
 		const promise = this.withCompositeProgress(this.viewletService.getProgressIndicator(viewletId), task, options);
 
-		// show activity bar
+		// show on activity bar
+		this.showOnActivityBar<P, R>(viewletId, options, promise);
+
+		return promise;
+	}
+
+	private withViewProgress<P extends Promise<R>, R = unknown>(viewId: string, task: (progress: IProgress<IProgressStep>) => P, options: IProgressCompositeOptions): P {
+
+		// show in viewlet
+		const promise = this.withCompositeProgress(this.viewsService.getProgressIndicator(viewId), task, options);
+
+		const location = this.viewDescriptorService.getViewLocation(viewId);
+		if (location !== ViewContainerLocation.Sidebar) {
+			return promise;
+		}
+
+		const viewletId = this.viewDescriptorService.getViewContainer(viewId)?.id;
```

Comment:
@eamodio can we ever have a view without container? I understand the previous condition because we can only show activity for viewlets and not panels, but this check I wonder...

Reply:
I dunno -- but the typings require those checks.